ReactiveCocoa/Swift/Property.swift

```diff
@@ -134,50 +132,21 @@ FIXME: DISABLED DUE TO COMPILER BUG
 /// The binding will automatically terminate when the property is deinitialized,
 /// or when the created signal sends a `Completed` event.
 public func <~ <T>(property: MutableProperty<T>, producer: SignalProducer<T, NoError>) -> Disposable {
-	let disposable = CompositeDisposable()
-	let propertyDisposable = property.producer.start(completed: {
-		disposable.dispose()
-	})
-
-	disposable.addDisposable(propertyDisposable)
-
+	var disposable: Disposable!
+	
 	producer.startWithSignal { signal, signalDisposable in
-		disposable.addDisposable(signalDisposable)
-
-		signal.observe(next: { [weak property] value in
-			property?.value = value
-			return
-		}, completed: {
-			disposable.dispose()
-		})
+		property <~ signal
```

Comment:
Although this implementation is much simpler (kudos!), there's unfortunately a bit of a problem right here.

If the property deinitializes, and the returned disposable is _not_ disposed, the producer's work may not be cancelled. In order for that to happen, we need to make sure that `signalDisposable` is disposed when `property.producer` completes.

The important bit is that it doesn't need to be put into any kind of serial or composite disposable, though.